src/blenderbim/blenderbim/bim/module/search/operator.py

```diff
@@ -381,3 +381,106 @@ def draw(self, context):
         row = self.layout.row(align=True)
         row.operator("bim.toggle_filter_selection", text="Select All").action = "SELECT"
         row.operator("bim.toggle_filter_selection", text="Deselect All").action = "DESELECT"
+
+
+class CreateIfcClassSelSet(bpy.types.Operator):
+    "Click to generate a selection set for all ifc classes"
+    bl_idname = "bim.create_ifcclass_ss"
+    bl_label = "Create IfcClass Selection Set"
+
+    def execute(self, context):
+        if "Selection Sets" in bpy.data.collections:
+            ss = bpy.data.collections["Selection Sets"]  
+        else:
+            ss = bpy.data.collections.new("Selection Sets")
+            ss.color_tag = "COLOR_04"
+
+        if "Ifc Classes" in bpy.data.collections:
+            ifc_classes = bpy.data.collections["Ifc Classes"]
+        else:
+            ifc_classes = bpy.data.collections.new("Ifc Classes")
+            ifc_classes.color_tag = "COLOR_04"
+            ss.children.link(ifc_classes)
+            
+        objects = context.selectable_objects
+        
+        for obj in objects:
+            ifc_element = tool.Ifc.get_entity(obj)
+            if not ifc_element:
+                continue     
+            
+            ifc_type = ifc_element.is_a()
+            if ifc_type in bpy.data.collections:
+                ifc_class = bpy.data.collections[ifc_type]
```

Comment:
Might be interesting to add a prefix or suffix to avoid name collisions, it may be safe to assume that users will be able to create collections named after ifc classes in other contexts.